src/test/java/io/confluent/idesidecar/websocket/proxy/ProxyContextTest.java

```diff
@@ -0,0 +1,180 @@
+package io.confluent.idesidecar.websocket.proxy;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import io.confluent.idesidecar.restapi.connections.CCloudConnectionState;
+import io.confluent.idesidecar.restapi.util.FlinkPrivateEndpointUtil;
+import io.quarkus.test.InjectMock;
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+import java.util.List;
+
+@QuarkusTest
+class ProxyContextTest {
+
+  @InjectMock
+  FlinkPrivateEndpointUtil flinkPrivateEndpointUtil;
+
+  private ProxyContext createContext() {
+    return new ProxyContext(
+        "conn1",
+        "us-west-2",  // Changed to match test private endpoints
+        "aws",
+        "env-123",
+        "org-456",
+        null
+    );
+  }
+
+  private ProxyContext createContextWithDifferentRegion() {
+    return new ProxyContext(
+        "conn1",
+        "eu-central-1",
+        "aws",
+        "env-123",
+        "org-456",
+        null
+    );
+  }
```

Comment:
nit: You could convert these methods to constants unless you plan to parameterize them:

```suggestion
  static final ProxyContext PROXY_CONTEXT_AWS_US_WEST = new ProxyContext(
      "conn1",
      "us-west-2",  // Changed to match test private endpoints
      "aws",
      "env-123",
      "org-456",
      null
  );
  
  static final ProxyContext PROXY_CONTEXT_AWS_EU_CENTRAL = new ProxyContext(
      "conn1",
      "eu-central-1",
      "aws",
      "env-123",
      "org-456",
      null
  );
```